In [1]:
import os
import sys; sys.path.append("~/HHbbgg_ETH_devel/Training/python") # to load packages
import training_utils as utils
reload(utils)
import preprocessing_utils as preprocessing
reload(preprocessing)
import optimization_utils as optimization
reload(optimization)
import numpy as np

Welcome to ROOTaaS 6.05/02


In [32]:
path = '/mnt/t3nfs01/data01/shome/nchernya/HHbbgg_ETH_devel/Training/output_files/txt_output/'
files = ['optJobs.txt','optJobs_2time.txt','local_optimzationWeekend.txt']
num = [50,200,50]

In [46]:
inp = open(path+files[0],"r")
dictionaries = []
for i in range(0,len(files)):
    lines = []
    inp = open(path+files[i],"r")
    for line in inp:
        dicti = {}
        if '(+/-' in line :
            lines.append(line)
            dicti['value'] = float(line[:line.find(' (')])
            dicti['params'] = line[:line.find(' }')]
            dicti['fromFile#'] = i
            if (len(lines)<num[i]+1) : dictionaries.append(dicti) #because I printed twice the output 
    inp.close()
        

In [65]:
from operator import itemgetter
print len(dictionaries)
newlist = sorted(dictionaries, key=itemgetter('value'), reverse=True)
#print newlist[0]
#print newlist
count=0
for dic in newlist:
    if dic['value']>0.5 : count+=1


300
145
[{'fromFile#': 1, 'params': "0.9647 (+/-0.0016) for {'reg_alpha': 0.0, 'learning_rate': 0.1, 'min_child_weight': 1e-05, 'n_estimators': 1000, 'reg_lambda': 0.01, 'max_depth': 10, 'gamma': 0}", 'value': 0.9647}, {'fromFile#': 1, 'params': "0.9645 (+/-0.0015) for {'reg_alpha': 0.0, 'learning_rate': 0.2, 'min_child_weight': 1e-06, 'n_estimators': 2000, 'reg_lambda': 0.1, 'max_depth': 15, 'gamma': 0}", 'value': 0.9645}, {'fromFile#': 1, 'params': "0.9643 (+/-0.0015) for {'reg_alpha': 0.0, 'learning_rate': 0.2, 'min_child_weight': 0.0001, 'n_estimators': 1500, 'reg_lambda': 0.1, 'max_depth': 15, 'gamma': 0}", 'value': 0.9643}, {'fromFile#': 1, 'params': "0.9643 (+/-0.0014) for {'reg_alpha': 0.0, 'learning_rate': 0.1, 'min_child_weight': 0.0001, 'n_estimators': 1500, 'reg_lambda': 0.01, 'max_depth': 10, 'gamma': 0}", 'value': 0.9643}, {'fromFile#': 1, 'params': "0.9641 (+/-0.0014) for {'reg_alpha': 0.0, 'learning_rate': 0.2, 'min_child_weight': 0.0001, 'n_estimators': 1000, 'reg_lamb

In [71]:
newlist

[{'fromFile#': 1,
  'params': "0.9647 (+/-0.0016) for {'reg_alpha': 0.0, 'learning_rate': 0.1, 'min_child_weight': 1e-05, 'n_estimators': 1000, 'reg_lambda': 0.01, 'max_depth': 10, 'gamma': 0}",
  'value': 0.9647},
 {'fromFile#': 1,
  'params': "0.9645 (+/-0.0015) for {'reg_alpha': 0.0, 'learning_rate': 0.2, 'min_child_weight': 1e-06, 'n_estimators': 2000, 'reg_lambda': 0.1, 'max_depth': 15, 'gamma': 0}",
  'value': 0.9645},
 {'fromFile#': 1,
  'params': "0.9643 (+/-0.0015) for {'reg_alpha': 0.0, 'learning_rate': 0.2, 'min_child_weight': 0.0001, 'n_estimators': 1500, 'reg_lambda': 0.1, 'max_depth': 15, 'gamma': 0}",
  'value': 0.9643},
 {'fromFile#': 1,
  'params': "0.9643 (+/-0.0014) for {'reg_alpha': 0.0, 'learning_rate': 0.1, 'min_child_weight': 0.0001, 'n_estimators': 1500, 'reg_lambda': 0.01, 'max_depth': 10, 'gamma': 0}",
  'value': 0.9643},
 {'fromFile#': 1,
  'params': "0.9641 (+/-0.0014) for {'reg_alpha': 0.0, 'learning_rate': 0.2, 'min_child_weight': 0.0001, 'n_estimators': 1

# Best set of parameters

{'fromFile#': 1, 'params': "0.9647 (+/-0.0016) for {'reg_alpha': 0.0, 'learning_rate': 0.1, 'min_child_weight': 1e-05, 'n_estimators': 1000, 'reg_lambda': 0.01, 'max_depth': 10, 'gamma': 0}", 'value': 0.9647}

In [52]:
import cPickle as pickle
outp = open(path+'bestParameters_28_05_2018.txt',"w")
for i in range(0,len(newlist)):
    outp.write(pickle.dumps(newlist[i]))
outp.close()

In [70]:
newdictionaries = []
for i in range(0,10):
    string = newlist[i]['params']
    dict = {}
    print string[string.find('{')-1:string.find('}')+1]
    dict = dict(string[string.find('{')-1:string.find('}')+1])
    newdictionaries.append(dict)

 {'reg_alpha': 0.0, 'learning_rate': 0.1, 'min_child_weight': 1e-05, 'n_estimators': 1000, 'reg_lambda': 0.01, 'max_depth': 10, 'gamma': 0}


TypeError: 'dict' object is not callable

In [ ]:
import json
path_txt = '/mnt/t3nfs01/data01/shome/nchernya/HHbbgg_ETH_devel/Training/output_files/txt_output/'
with open(path_txt+'first10results.json', 'w') as fout:
    json.dump(your_list_of_dict, fout)